### PR TITLE
feature TSIG configuration

### PR DESCRIPTION
**Pull request by eumel8**
To exchange DNS zones between two bind installations like master/slave you can sign this transaction with a key. It's more secure as IP-based zone transfer and protect against spoofing (see ftp://ftp.isc.org/isc/bind/9.8.0-P4/doc/arm/Bv9ARM.ch04.html#tsig)

### CODE REVIEW

File: templates/tsig.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+key <%= @keyname %>. {
+  algorithm <%= @algorithm %>;
+  secret "<%= @secret %>";
+};
+<% if server.class == Array -%>
```
Comment (solarkennedy) [inline templates/tsig.erb]: `@server` ?

**Comment by solarkennedy** [comment]
Can you add a test for this defined type to ensure that the erb syntax is valid?

Maybe do at least two contexts, one with an array of ips for server and one with a single server ip?

**Comment by eumel8** [comment]
@solarkennedy I added some simple checks. Maybe it's enough for the first try.

File: templates/tsig.erb (hunk 1 above)
Comment (solarkennedy) [inline templates/tsig.erb]: Use `@server` notation in erb variables

**Comment by eumel8** [comment]
overseen

**Comment by solarkennedy** [comment]
Thanks!

**Pull request merged**